### ISSUE TITLE
Final Proofreading: Conventions and Definitions

### ISSUE THREAD

**Issue by samuel-lucas6** [cfrg/draft-irtf-cfrg-aegis-aead#74]
### Manual Review
https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-aegis-aead-14#section-2-3
`Primitives:` => Rename to `Operations`? Or split up the first/second parts, like in the AEGIS paper.

https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-aegis-aead-14#section-2-4.6.1
`the remainder of the Euclidean division between a as the dividend and b as the divisor` => Could be more developer friendly but not wrong.

`mod` => could be changed to `%` operator to match the others.

https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-aegis-aead-14#section-2-4.8.1
`padding operation. Trailing zeros are concatenated to x until the total length is a multiple of n bits` => `Returns x padded with zeros until the total length is a multiple of n bits` or `The result of padding x with zeros`...

**Also, it should be clear that no padding should happen if x is empty (e.g., empty ad or msg doesn't get padded/absorbed into the state)**.

https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-aegis-aead-14#section-2-4.9.1
`truncation operation. The first n bits of x are kept.` => `Returns the first n bits of x`.

My point with these is that I think they should be written in a consistent style.

https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-aegis-aead-14#section-2-4.10.1
`splitting operation. x is split into n-bit blocks, ignoring partial blocks.` => `Returns x split into n-bit blocks, ignoring partial blocks`. Could maybe reword 'partial blocks' part.

https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-aegis-aead-14#section-2-4.12.1
`ShiftRows, MixColums and AddRoundKey` => Add a final comma.

`section 5` => `Section 5`?

https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-aegis-aead-14#section-2-4.14.1
`CtEq()` => The other function names aren't abbreviated, although this is tricky due to the lengthy names.

`an exact match, False otherwise` => `and False otherwise`.

https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-aegis-aead-14#section-2-5
`ai, xi, ci, cn` vs `key, nonce, ad_len_bits, msg_len_bits` naming => `ad_block`, `msg_block`, `enc_block` or similar? However, the shortened does distinguish internal from external parameters.

Could reorder `Update()` to be after `Init()` to match the sections later on.

https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-aegis-aead-14#section-2-7 and https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-aegis-aead-14#section-2-12.1.1
Do we want to mention AEGIS-128X/256X in these places?

### Grammarly
https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-aegis-aead-14#section-2-4.12.1
Agrees with me about the comma.

https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-aegis-aead-14#section-2-9.1.1
Complains about `i-th/j-th`. This is fine.

**Comment by jedisct1** [comment]
> ai, xi, ci, cn vs key, nonce, ad_len_bits, msg_len_bits naming => ad_block, msg_block, enc_block or similar? However, the shortened does distinguish internal from external parameters.

Yes, having.short names for internal variables improves readability and is closer to what actual implementations are likely to do, IMHO.

**Comment by jedisct1** [comment]
> Could reorder Update() to be after Init() to match the sections later on.

Done.

But what's a bit annoying now, and why `Update` was initially first, is that other functions can't be implemented without the `Update` function.

With `Update` first, we can a sequence of functions that can be implemented in order (and `Finalize` is at the end because it needs `Absorb`).

**Comment by jedisct1** [comment]
> Do we want to mention AEGIS-128X/256X in these places?

Too complicated, as it depends on the parallelism degree, and that's something we haven't defined at that point.

**Comment by samuel-lucas6** [comment]
> With Update first, we can a sequence of functions that can be implemented in order (and Finalize is at the end because it needs Absorb).

So do you think the `The Update Function` sections should be moved before `The Init Function` sections? I see what you mean but think it currently works.

**Comment by jedisct1** [comment]
Having the `Update` function before `Init` would indeed be better, and it used to be the case.

The was changed in 69cc660bc233cb94099aa4cea7de18d7d4421437

I'd like to revert that change, but maybe in a further revision that barely changes anything else.

**Comment by jedisct1** [comment]
Having `Update` first is not just for C implementations, but also to incrementally check the test vectors.

The test vectors for `Update` are the first ones that can be checked. It's not worth implementing any other function before they pass.

**Comment by samuel-lucas6** [comment]
I'm happy for that change to be reverted for the reasons you outline.

**Comment by jedisct1** [comment]
> I'm happy for that change to be reverted for the reasons you outline.

I'd rather wait for the next revision, as even though sections are just swapped, that would appear as a really big change.
